src/core/thread/mle.cpp

```diff
@@ -1625,6 +1625,11 @@ ThreadError Mle::SendDataRequest(const Ip6::Address &aDestination, const uint8_t
     else
     {
         SuccessOrExit(error = SendMessage(*message, aDestination));
+
+        if ((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0)
+        {
+            mNetif.GetMeshForwarder().GetDataPollManager().SetAttachMode(true);
```

Comment:
@jwhui I suggest that we use `DataPollManager::HandleResponseExpected()` here instead of `SetAttachMode()`, then we don't need to do anything from `HandleDataResponse()` and even if we don't get a response the fast data polls will stop after some number of attempts.